projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-trait.php

```diff
@@ -26,18 +26,162 @@ trait Jetpack_WooCommerce_Analytics_Trait {
+
+		if (
+			false !== $transient_value &&
+			! empty( $transient_value['checkout_content_source'] ) &&
+			! empty( $transient_value['cart_content_source'] )
+		) {
+			$this->cart_content_source     = $transient_value['cart_content_source'];
+			$this->checkout_content_source = $transient_value['checkout_content_source'];
+			return;
+		}
+
+		// Cart/Checkout *pages* are in use if the templates are not in use. Return their content and do nothing else.
+		if ( ! $this->cart_checkout_templates_in_use ) {
+			$cart_page_id     = wc_get_page_id( 'cart' );
+			$checkout_page_id = wc_get_page_id( 'checkout' );
+			$cart_page        = get_post( $cart_page_id );
+			$checkout_page    = get_post( $checkout_page_id );
+
+			if ( $cart_page && isset( $cart_page->post_content ) ) {
+				$this->cart_content_source = $cart_page->post_content;
+			}
+			if ( $checkout_page && isset( $checkout_page->post_content ) ) {
+				$this->checkout_content_source = $checkout_page->post_content;
+			}
+
+			set_transient(
+				$cart_checkout_content_cache_transient_name,
+				array(
+					'cart_content_source'     => $this->cart_content_source,
+					'checkout_content_source' => $this->checkout_content_source,
+				),
+				DAY_IN_SECONDS
+			);
+			return;
+		}
+
+		// We are in a Block theme - so we need to find out if the templates are being used.
+		if ( function_exists( 'get_block_template' ) ) {
+			$checkout_template = get_block_template( 'woocommerce/woocommerce//page-checkout' );
+			$cart_template     = get_block_template( 'woocommerce/woocommerce//page-cart' );
+			if ( ! $checkout_template ) {
+				$checkout_template = get_block_template( 'woocommerce/woocommerce//checkout' );
+			}
+			if ( ! $cart_template ) {
+				$cart_template = get_block_template( 'woocommerce/woocommerce//cart' );
+			}
+		}
+
+		if ( function_exists( 'gutenberg_get_block_template' ) ) {
+			$checkout_template = gutenberg_get_block_template( 'woocommerce/woocommerce//page-checkout' );
+			$cart_template     = gutenberg_get_block_template( 'woocommerce/woocommerce//page-cart' );
+			if ( ! $checkout_template ) {
+				$checkout_template = gutenberg_get_block_template( 'woocommerce/woocommerce//checkout' );
+			}
+			if ( ! $cart_template ) {
+				$cart_template = gutenberg_get_block_template( 'woocommerce/woocommerce//cart' );
+			}
+		}
+
+		if ( ! empty( $checkout_template->content ) ) {
+			// Checkout template is in use, but we need to see if the page-content-wrapper is in use, or if the template is being used directly.
+			$this->checkout_content_source = $checkout_template->content;
+			$is_using_page_content         = str_contains( $checkout_template->content, '<!-- wp:woocommerce/page-content-wrapper {"page":"checkout"}' );
+
+			if ( $is_using_page_content ) {
+				// The page-content-wrapper is in use, so we need to get the page content.
+				$checkout_page_id = wc_get_page_id( 'checkout' );
+				$checkout_page    = get_post( $checkout_page_id );
```

Comment:
Nitpic: Similar to my other suggestions, I wonder if we should simplify this part as well:

```suggestion
				$checkout_page    = get_post( wc_get_page_id( 'checkout' ) );
```